databricks_cli/stack/api.py

```diff
@@ -459,56 +472,115 @@ def _validate_config(self, stack_config):
         :return: None. Raises errors to stop deployment if there is a problem.
         """
-        if STACK_NAME not in stack_status:
-            raise StackError("'{}' not in status.".format(STACK_NAME))
-        if STACK_RESOURCES not in stack_status:
-            raise StackError("'{}' not in status".format(STACK_RESOURCES))
-        if STACK_DEPLOYED not in stack_status:
-            raise StackError("'{}' not in status".format(STACK_DEPLOYED))
-        for deployed_resource in stack_status.get(STACK_DEPLOYED):
-            if RESOURCE_ID not in deployed_resource:
-                raise StackError("{} doesn't exist in deployed resource status".format(
-                    RESOURCE_ID))
-            if RESOURCE_SERVICE not in deployed_resource:
-                raise StackError("{} doesn't exist in deployed resource status".format(
-                    RESOURCE_SERVICE))
-            if RESOURCE_PHYSICAL_ID not in deployed_resource:
-                raise StackError("{} doesn't exist in deployed resource status".format(
-                    RESOURCE_PHYSICAL_ID))
+        self._assert_field_in_stack_status(STACK_NAME, stack_status)
+        self._assert_field_in_stack_status(STACK_RESOURCES, stack_status)
+        self._assert_field_in_stack_status(STACK_DEPLOYED, stack_status)
+
+        for resource_status in stack_status.get(STACK_DEPLOYED):
+            self._assert_field_in_resource_status(RESOURCE_ID, resource_status)
+            self._assert_field_in_resource_status(RESOURCE_SERVICE, resource_status)
+            self._assert_field_in_resource_status(RESOURCE_PHYSICAL_ID, resource_status)
+            self._assert_field_in_resource_status(RESOURCE_DEPLOY_OUTPUT, resource_status)
+
+            resource_id = resource_status[RESOURCE_ID]
+            resource_service = resource_status.get(RESOURCE_SERVICE)
+            resource_physical_id = resource_status.get(RESOURCE_PHYSICAL_ID)
+
+            if resource_service == JOBS_SERVICE:
+                self._assert_field_in_resource_physical_id(JOBS_RESOURCE_ID, resource_physical_id,
+                                                           resource_service, resource_id)
+            elif resource_service == WORKSPACE_SERVICE:
+                self._assert_field_in_resource_physical_id(WORKSPACE_RESOURCE_PATH,
+                                                           resource_physical_id,
+                                                           resource_service, resource_id)
+            elif resource_service == DBFS_SERVICE:
+                self._assert_field_in_resource_physical_id(DBFS_RESOURCE_PATH, resource_physical_id,
+                                                           resource_service, resource_id)
+            else:
+                raise StackError("{} not a valid resource status service".format(resource_service))
+
+    def _assert_field_in_resource_physical_id(self, field, physical_id, service, resource_id):
```

Comment:
There are too many small helpers, which may be an overhead to maintain in the long run. Let's use some generic methods and make simple messages.